app/javascript/src/components/Player.tsx

```diff
@@ -349,28 +351,42 @@ class Player extends React.Component<PlayerProps, PlayerState> {
     )
   }
 
+  private strip(html: string){
+    var doc = new DOMParser().parseFromString(html, 'text/html');
+    return doc.body.textContent || "";
+  }
+
   private translate(text: string) {
     const tags = text.match(/\{\{(.+?)\}\}/g)
-
     if (tags) {
       tags.forEach(tag => {
         const tagName = tag.replace(/[\{\}]/g, '')
+        var cleanedTag = this.strip(tagName)
 
-        const relevantStat = this.state.currentStats.find(stat => {
-          return stat.name == tagName
+        const operands = cleanedTag.split(/[+*/-]/g)
+        const statArray = operands.map(item => {
+          return this.findStat(item.trim())
         })
+        statArray.forEach(stat => {
+          if (stat) {
+            cleanedTag = cleanedTag.replace(new RegExp(stat.name), stat.value.toString())
+          }
```

Comment:
looks like the parser lib lets you pass in a variables hash to swap in, thoughts on using that instead of manually finding and replacing?

Reply:
although, I suppose this tactic gives you a good spot to identify invalid variables and gracefully back out of the whole process